### ISSUE TITLE
Using the same attachment as a color & input attachment for a subpass triggers questionable warnings

### ISSUE THREAD

**Issue by DominikWitczakAMD** [KhronosGroup/Vulkan-LoaderAndValidationLayers#1080]
Consider an app which uses a renderpass attachment X both as a color & an input attachment in the same subpass. This triggers the following warnings:

![image](https://cloud.githubusercontent.com/assets/17761066/19597860/80637d0a-9797-11e6-83ba-e975253d429d.png)

![image](https://cloud.githubusercontent.com/assets/17761066/19597891/aa5b07ae-9797-11e6-9766-00cdfbb7e6b2.png)

However:

* The attachment cannot be in COLOR_ATTACHMENT_OPTIMAL layout because that's illegal as per spec.
* The attachment cannot be in READ_ONLY_OPTIMAL layout because of the same reason.
* The attachment cannot be in two layouts at once :)

**Comment by chrisforbes** [comment]
Yep, that's pretty awful. To do this properly we need to combine all the constraints on the layout, rather than making a fuss about each constraint individually.
